app/cdap/components/ColumnInsights/Components/ColumnDetails/index.tsx

```diff
@@ -0,0 +1,194 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+
+const PREFIX = 'features.WranglerNewUI.ColumnInsights';
+interface IColumnDetailsProps {
+  columnName: string;
+  characterCount: string;
+  distinctValues: number;
+  dataTypeString: string;
+  renameColumnNameHandler: (oldColumnName: string, newColumnName: string) => void;
+  dataTypeHandler: (dataType: string) => void;
+  columnType: string;
+  columnHeaderList: string[];
+}
+
+const ColumnDetailsContainer = styled(Box)`
+  padding-bottom: 20px;
+  border-bottom: 1px solid ${grey[300]};
+`;
+
+const CustomizedColumnNameEditBox = styled(Box)`
+  display: flex;
+  gap: 16px;
+`;
+
+const CustomizedIconButton = styled(IconButton)`
+  cursor: pointer;
+  justify-content: flex-start !important;
+`;
+
+const ColumnInsightsDetailsBox = styled(Box)`
+  margin-top: 20px;
+`;
+
+const ColumnInsightsDetailsCountBox = styled(Box)`
+  display: flex;
+  gap: 27px;
+  align-items: center;
+  margin-bottom: 7px;
+`;
+
+export default function({
+  columnName,
+  characterCount,
+  distinctValues,
+  dataTypeString,
+  renameColumnNameHandler,
+  dataTypeHandler,
+  columnType,
+  columnHeaderList,
+}: IColumnDetailsProps) {
+  const defaultValueProvided =
+    DATATYPE_OPTIONS &&
+    Array.isArray(DATATYPE_OPTIONS) &&
+    DATATYPE_OPTIONS?.length &&
+    DATATYPE_OPTIONS.filter((each) => each.value === columnType?.toLowerCase());
+  const [dataTypeValue, setDataTypeValue] = useState<string>();
+  const [canEdit, setCanEdit] = useState(false);
+  const [inputValue, setInputValue] = useState(columnName);
+  const [errorMessage, setErrorMessage] = useState({
+    hasError: false,
+    message: '',
+  });
+
+  const displayMessage = {
+    invalidMessage: `${PREFIX}.error.invalidError`,
+    matchedColumnMessage: `${PREFIX}.error.matchedColumnError`,
+  };
+
+  const checkForInvalidInput = (renamedString: string) => {
+    if (renamedString === '' || !/^\w+$/.test(renamedString)) {
+      setErrorMessage({
+        hasError: true,
+        message: displayMessage.invalidMessage,
+      });
+    } else if (columnHeaderList.includes(renamedString) && columnName !== renamedString) {
+      setErrorMessage({
+        hasError: true,
+        message: displayMessage.matchedColumnMessage,
+      });
+    } else {
+      setErrorMessage({
+        hasError: false,
+        message: '',
+      });
+    }
+  };
+
+  const handleDataTypeChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setDataTypeValue(e.target.value);
+    dataTypeHandler(e.target.value);
+  };
+
+  const editHandler = () => {
+    setCanEdit(true);
+  };
+
+  const onChangeHandler = (e: React.ChangeEvent<HTMLInputElement>) => {
+    checkForInvalidInput(e.target.value);
+    setInputValue(e.target.value);
+  };
+
+  const onEnter = (e: React.KeyboardEvent<HTMLElement>) => {
+    if (
+      (e.target as HTMLInputElement).value !== columnName &&
+      !errorMessage.hasError &&
+      e.keyCode === 13
```

Comment:
Please use `KeyboardEvent.key` or `KeyboardEvent.code`. This would be `e.key === 'Enter'`. `keyCode` is deprecated.